src/app/clusters/device-energy-management-server/device-energy-management-server.cpp

```diff
@@ -654,6 +746,96 @@ void Instance::HandleRequestConstraintBasedForecast(HandlerContext & ctx,
         return;
     }
 
+    uint32_t currentUtcTime = 0;
+    status                  = GetCurrentUtcTime(currentUtcTime);
+    if (status != Status::Success)
+    {
+        ChipLogError(Zcl, "DEM: Forecast is Null");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+        return;
+    }
+
+    // Check for invalid power levels and whether the constraint time/duration is in the past
+    {
+        auto iterator = constraints.begin();
+        if (iterator.Next())
```

Comment:
Again, where is the required-when-iterating-lists status check?

Reply:
I have added a check.